modules/core/include/opencv2/core/hal/intrin_neon.hpp

```diff
@@ -1969,11 +1969,23 @@ inline v_int32x4 v_round(const v_float32x4& a)
 #else
 inline v_int32x4 v_round(const v_float32x4& a)
 {
-    static const int32x4_t v_sign = vdupq_n_s32(1 << 31),
-        v_05 = vreinterpretq_s32_f32(vdupq_n_f32(0.5f));
+    static const float32x4_t v0_0_f32  = vdupq_n_f32(0.0);
```

Comment:
@Kumataro, can you please fix HAL implementation of v_round as well?

Reply:
@vpisarev 

OK, I fixed it and committed. 
However, I ran `opencv_test_core`, there are 1 failed.
I believe it doesn't used `v_round()`.I think this is a potential problem.

```plaintext
[----------] Global test environment tear-down
[ SKIPSTAT ] 86 tests skipped
[ SKIPSTAT ] TAG='mem_2gb' skip 1 tests
[ SKIPSTAT ] TAG='skip_other' skip 85 tests
[==========] 11949 tests from 264 test cases ran. (351138 ms total)
[  PASSED  ] 11948 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] Core/HasNonZeroLimitValues.hasNonZeroLimitValues/1, where GetParam() = (5, 320x240)
```

```
Note: Google Test filter = Core/HasNonZeroLimitValues*
[==========] Running 8 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 8 tests from Core/HasNonZeroLimitValues
[ RUN      ] Core/HasNonZeroLimitValues.hasNonZeroLimitValues/0, where GetParam() = (5, 1x1)
[       OK ] Core/HasNonZeroLimitValues.hasNonZeroLimitValues/0 (0 ms)
[ RUN      ] Core/HasNonZeroLimitValues.hasNonZeroLimitValues/1, where GetParam() = (5, 320x240)
/home/kmtr/work/opencv/modules/core/test/test_hasnonzero.cpp:109: Failure
Value of: hasNonZero(m)
  Actual: false
Expected: true
[  FAILED  ] Core/HasNonZeroLimitValues.hasNonZeroLimitValues/1, where GetParam() = (5, 320x240) (1 ms)
[ RUN      ] Core/HasNonZeroLimitValues.hasNonZeroLimitValues/2, where GetParam() = (5, 127x113)
[       OK ] Core/HasNonZeroLimitValues.hasNonZeroLimitValues/2 (0 ms)
[ RUN      ] Core/HasNonZeroLimitValues.hasNonZeroLimitValues/3, where GetParam() = (5, 1x113)
[       OK ] Core/HasNonZeroLimitValues.hasNonZeroLimitValues/3 (1 ms)
[ RUN      ] Core/HasNonZeroLimitValues.hasNonZeroLimitValues/4, where GetParam() = (6, 1x1)
[       OK ] Core/HasNonZeroLimitValues.hasNonZeroLimitValues/4 (0 ms)
[ RUN      ] Core/HasNonZeroLimitValues.hasNonZeroLimitValues/5, where GetParam() = (6, 320x240)
[       OK ] Core/HasNonZeroLimitValues.hasNonZeroLimitValues/5 (2 ms)
[ RUN      ] Core/HasNonZeroLimitValues.hasNonZeroLimitValues/6, where GetParam() = (6, 127x113)
[       OK ] Core/HasNonZeroLimitValues.hasNonZeroLimitValues/6 (0 ms)
[ RUN      ] Core/HasNonZeroLimitValues.hasNonZeroLimitValues/7, where GetParam() = (6, 1x113)
[       OK ] Core/HasNonZeroLimitValues.hasNonZeroLimitValues/7 (0 ms)
[----------] 8 tests from Core/HasNonZeroLimitValues (4 ms total)

[----------] Global test environment tear-down
[==========] 8 tests from 1 test case ran. (5 ms total)
[  PASSED  ] 7 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] Core/HasNonZeroLimitValues.hasNonZeroLimitValues/1, where GetParam() = (5, 320x240)

 1 FAILED TEST
```

Reply:
OK, I cleared it is not related with this work.
(Some code is updated, however almost are not changed.)
I believe we can split from  this issue.

https://github.com/opencv/opencv/blob/953dddd26b5cdc32fc9c2fc21fdedef1fa6fb04d/modules/core/src/has_non_zero.simd.hpp#L192-L202

I disabled L196 to stop using simd, same this test is passed.
And those code is simple, loading from memory and or-ing registers, checking to equal it with zero. So this logic is not related currently fixing. 

More investigation is required, but it is related with only special cases(denorm_min).

https://github.com/opencv/opencv/blob/953dddd26b5cdc32fc9c2fc21fdedef1fa6fb04d/modules/core/test/test_hasnonzero.cpp#L86-L110